hydra/_internal/core_plugins/fish_completion.py

```diff
@@ -22,8 +19,14 @@ def install(self) -> None:
     set -l parts (string split -n ' ' $COMP_LINE)
     if test "$parts[1]" = "python" -o "$parts[1]" = "python3"
         set cmd "$parts[1] $parts[2]"
+        set script "$parts[2]"
     else
         set cmd "$parts[1]"
+        set script "$parts[1]"
+    end
+
+    if not grep -q "@hydra.main" $script
```

Comment:
this should only be used for "python foo.py" mode.
if you are installing an app with entry point scripts you will end up with an executable (.exe on Windows).

Reply:
Nice catch.